kb/prometheus-grafana-setup.md

```diff
@@ -0,0 +1,247 @@
+   depends_on:
+     - temporal
+#...
+```
+
+In this example, we use the Prometheus docker image and the default ports for the container, and set the path to a Prometheus configuration file for the docker container. For other ways to set your Prometheus configuration, see the [Prometheus Configuration documentation](https://prometheus.io/docs/prometheus/latest/configuration/configuration/).
+
+Configure your Prometheus setup to scrape metrics data from the Temporal Cluster and SDK Client target endpoints.
+
+For example, in your local docker-compose Temporal Cluster, you can create a Prometheus configuration file (in this example, at ./deployment/prometheus/config.yml) to scrape metrics from targets set on the docker-compose Temporal Cluster, and SDK Clients from the previous examples:
+
+```
+global:
+ scrape_interval: 10s
+scrape_configs:
+ - job_name: 'temporalmetrics'
+   metrics_path: /metrics
+   scheme: http
+   static_configs:
+     # Cluster metrics target
+     - targets:
+         - 'host.docker.internal:8000'
+       labels:
+         group: 'server-metrics'
+
+     # Local app targets (if configured)
+     - targets:
+         - 'host.docker.internal:8077'
+         - 'host.docker.internal:8078'
+       labels:
+         group: 'sdk-metrics'
+```
+
+In this example, we have a YAML configuration file setting Prometheus to scrape at 10 second interval, with static targets to listen for Cluster metrics on `host.docker.internal:8000` and SDK metrics on two targets `host.docker.internal:8077` and `host.docker.internal:8078` for `WorkflowServiceStubs` that can be used to create Workers and make Client API calls to start Workflow Executions, send Signals, Queries etc.
+You can set up as many targets as required.
+Note that these SDK metrics ports must be set in your application and are discussed in the [SDK Metrics](#sdkmetricssetup) section for details.
+
+See [Prometheus configuration](https://prometheus.io/docs/prometheus/latest/configuration/configuration/) for more information.
+
+To check whether you’re receiving your metrics data, start your local docker-compose Temporal Cluster with the configuration provided in the examples here) and go to
+
+- http://localhost:8000/metrics - this is the port you’re exposing your Cluster metrics on. You should see all the server metrics emitted when you start up your local docker-compose Temporal Cluster.
+- http://localhost:8077/metrics - this is the port you’re exposing your SDK metrics on. Depending on whether you have set this port on the Client that is starting your Worker or your Workflow Executions, the related metrics should show when you start your Worker or Workflow Execution.
+- http://localhost:8078/metrics - this is the port you’re exposing your SDK metrics on. Depending on whether you have set this port on the Client that is starting your Worker or your Workflow Executions, the related metrics should show when you start your Worker or Workflow Execution.
```

Comment:
These are the same explanation?

Reply:
Yes. Depending on what you assign the ports to in your app code, you'll see the relevant metrics.